doc/classes/File.xml

```diff
@@ -486,6 +486,7 @@
 			</argument>
 			<description>
 				Stores any Variant value in the file. If [code]full_objects[/code] is [code]true[/code], encoding objects is allowed (and can potentially include code).
+				[b]Note:[/b] When encoding objects, only properties that are configured as storable will be included. By default, script variables are configured as [constant PROPERTY_USAGE_SCRIPT_VARIABLE], which is [b]not[/b] included. You can specify the usage of a property by overriding the [method Object._get_property_list] method in your class. See [enum PropertyUsageFlags] for the possible usage flags.
```

Comment:
> only properties that are configured as storable will be included

It's not clear what "configured as storable" means.

> By default, script variables are configured as [constant PROPERTY_USAGE_SCRIPT_VARIABLE], which is [b]not[/b] included.

Note that for example exported property like `export var something` has more flags set, including `PROPERTY_USAGE_STORAGE` and thus it would be stored.

I'm not sure what flags are set for what properties but for sure info added in here needs to be correct.

Reply:
It's way better but I think that mentioning `PROPERTY_USAGE_SCRIPT_VARIABLE` is not necessary at all. It just adds unrelevant information, it's kinda a noise. Saying that `PROPERTY_USAGE_STORAGE` flag needs to be set and listing which properties/variables **for sure** have / don't have `PROPERTY_USAGE_STORAGE` flag set by default should be enough. But again: someone else knowing the subject better will need to verify that info (for what this flag is set / not set). I'm not 100% sure about these. You know, maybe there's some special case e.g. like maybe script variables for `Resource`s are treated differently or something like that, I don't know.

As a side note: make sure to squash your commits into one, [more info here](https://docs.godotengine.org/en/3.3/community/contributing/pr_workflow.html#modifying-a-pull-request).

Reply:
Thanks for the feedback!

I think removing the bit about `PROPERTY_USAGE_SCRIPT_VARIABLE` makes a lot of sense. Also, if we remove that, then it might be overkill to enumerate all the use-cases that result in `PROPERTY_USAGE_STORAGE` being set or not? I also have no idea what the possible cases are, and I also don't know whether they are likely to change in the future, making any description we add here obsolete.

And thanks for the squash reminder! Done.

Reply:
Rather than try to list use-cases, I just added a sentence explaining that you can just check how properties are being configured. WDYT?

Reply:
True, listing all might be an overkill. So just pointing in the right direction might be indeed enough. The only thing I'd point out is that this sentence:
>You can specify the usage of a property by overriding the [method Object._get_property_list] method in your class.

isn't totally correct. For example this:
```
export(String) var test

func _get_property_list() -> Array:
	return [
		{
			name = "test",
			type = TYPE_STRING,
			usage = 0,
		},
	]
```
will result in `test` still being stored (`get_property_list()` would list `test` twice with `usage` flags 8199 and 0 respectively). So I'm guessing that by overriding `_get_property_list()` you can make some property storable (by setting `PROPERTY_USAGE_STORAGE` flag) but you can't make a property which is storable by default un-storable. Hope this makes sense. So maybe you can just add flags but not remove. But as always, I'm not sure. :)

Reply:
Ahh, interesting... Yeah, I've no idea what the behavior should be, and I don't have the time to try actually changing any underlying logic, so I think just modifying the comment to only mention adding usage flags is the way to go!

I've re-worded this again.

I didn't update the comment to mention the duplication of usage configurations, since we aren't necessarily certain how/why/when this will happen, or whether this will change in the near future. And I think it would distract from the rest of the doc comment.

(Sorry about the delay, I took the weekend off for GMTK!)

Reply:
It makes sense what you're saying... but are you sure you didn't mess up something when commiting? Seems like some previous version/commit mentioning "storable" and `PROPERTY_USAGE_SCRIPT_VARIABLE`.

Reply:
Argh! Yeah, I'm definitely still figuring out the PR rebase workflow.

I think it should be fixed now!

Reply:
Yeah, seems fine!